Updating build_info.json of currency packages, to include default branch info.

Ryan-Dmello opened the pull request:
## Checklist
<!--- Goto Preview tab for better readability -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

nikhil-kalbande commented:
@Ryan-Dmello  We will need to add default_branch here https://github.com/ppc64le/build-scripts/blob/master/script/validate_builds.py#L104 as we expect it is mandatory field. This will help to avoid missing it in further PR's

Ryan-Dmello commented:
> @Ryan-Dmello We will need to add default_branch here https://github.com/ppc64le/build-scripts/blob/master/script/validate_builds.py#L104 as we expect it is mandatory field. This will help to avoid missing it in further PR's

Updated the build-info.json generation logic and added the requested params as mandatory

seth-priya commented on a/afero/build_info.json:
```diff
@@ -2,6 +2,8 @@
     "package_name" : "afero",
     "github_url":"https://github.com/spf13/afero",
     "version": "v1.9.3",
+    "default_branch": "master",
```
Remind me - the reason why we need to set this in the build_info files rather than fetch it programmatically is because of the concern around the number of GitHub API calls, correct?

Ryan-Dmello replied:
Correct.

seth-priya commented on i/influxdb/build_info.json:
```diff
@@ -2,6 +2,8 @@
     "version": "v2.5.1",
+    "default_branch": "master",
+    "build_script" : "influxdb_ubi_8.5.sh",
     "package_dir": "i/influxdb/Dockerfiles/",
```
I know this is not part of your changes, but just wondering, shouldn't the package_dir here be the top level directory "i/influxdb" as is the case for other packages?

Ryan-Dmello replied:
Yes, `package_dir` should be a top-level directory. But looks like in current case, it used for docker command.

Ryan-Dmello commented on a/apache-nifi/build_info.json:
```diff
@@ -2,6 +2,8 @@
     "github_url":"https://github.com/apache/nifi",
     "version": "rel/nifi-1.17.0",
+    "default_branch": "main",
+    "build_script" : "apache-nifi_1.17.0_ubi_8.5.sh",
```
Line 17 is for any tags, related to generic release versions (branches matching `rel/nifi-`). In case we want to run for any branch line 6 script will be used.
